samples/net/mqtt_publisher/src/main.c

```diff
@@ -61,15 +61,41 @@ struct mqtt_client_ctx {
 	void *publish_data;
 };
 
-/* This is the network context structure. */
-static struct net_context *net_ctx;
-
 /* The mqtt client struct */
 static struct mqtt_client_ctx client_ctx;
 
 /* This routine sets some basic properties for the network context variable */
-static int network_setup(struct net_context **net_ctx, const char *local_addr,
-			 const char *server_addr, u16_t server_port);
+static int network_setup(void);
+
+#if defined(CONFIG_MQTT_LIB_TLS)
+/* TLS */
+#define TLS_SNI_HOSTNAME "localhost"
+#define TLS_REQUEST_BUF_SIZE 1500
+#define TLS_PRIVATE_DATA "Zephyr TLS mqtt publisher"
+
+#if !defined(CONFIG_NET_APP_TLS_STACK_SIZE)
+#define CONFIG_NET_APP_TLS_STACK_SIZE 7200
+#endif
+
+static u8_t tls_request_buf[TLS_REQUEST_BUF_SIZE];
+
+NET_STACK_DEFINE("mqtt_tls_stack", tls_stack,
+		CONFIG_NET_APP_TLS_STACK_SIZE, CONFIG_NET_APP_TLS_STACK_SIZE);
+
+NET_APP_TLS_POOL_DEFINE(tls_mem_pool, 30);
+
+int setup_cert(struct net_app_ctx *ctx, void *cert)
+{
+	/* TODO: PSK and CA verification */
```

Comment:
Before TLS support patch can be accepted, this part of the code needs to be implemented too.